### PR TITLE
[MNG-6713] Fix ExclusionArtifactFilter to respect wildcard exclusions.

### PR DESCRIPTION
**Pull request by saturnism**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MNG-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MNG-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Comment by michael-o** [comment]
@saturnism Can you change the code as requested by @eolivelli?

File: maven-core/src/main/java/org/apache/maven/artifact/resolver/filter/ExclusionArtifactFilter.java [hunk 1]
```diff
@@ -41,8 +41,20 @@ public boolean include( Artifact artifact )
     {
         for ( Exclusion exclusion : exclusions )
         {
-            if ( exclusion.getGroupId().equals( artifact.getGroupId() )
-                    && exclusion.getArtifactId().equals( artifact.getArtifactId() ) )
+            if ( "*".equals( exclusion.getGroupId() ) && "*".equals( exclusion.getArtifactId() ) )
```
[the remark this answers is not in the document]
Comment (saturnism) [reply maven-core/src/main/java/org/apache/maven/artifact/resolver/filter/ExclusionArtifactFilter.java]: moved it to a private constant. unless if there is a more suitable place for a public global constant, I feel keeping it local is better.

**Comment by Tibor17** [comment]
@michael-o 
@olamy 
@eolivelli 
@rfscholte 
After Robert merged the comit to remote master, the build failed with strange issue on sockets.
Where these sockets come from? The MRM plugin?
https://builds.apache.org/job/maven-box/job/maven/job/master/269/testReport/junit/org.apache.maven.it/MavenITmng4554PluginPrefixMappingUpdateTest/windows_jdk12___Run_ITs_Windows_Java_12___testitCached/
Do you check also the Jenkins/master after pushed to master?

**Comment by rfscholte** [comment]
Of course I check master and I get notifications. The commit succeeded as a branch.
Based on the error it is very unlikely it is a Windows + JDK12 issue. I'll give it another try.

**Comment by Tibor17** [comment]
@rfscholte 
I doubt about running the build again should be the right way to make the build passed because it looks like there is no repo on server socket on `localhost:80`. The question is where this come from.
Could not transfer metadata org.apache.maven.its.mng4554/maven-metadata.xml from/to central (http://localhost:-1/repo-1): Connect to localhost:80 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused: connect

**Comment by Tibor17** [comment]
@rfscholte 
I have fixed the build 269. The error is mentioned in JavaDoc
```
    /**
     * @return The actual port the connector is listening on or -1 if there 
     * is no port or the connector is not open.
     */
    int getLocalPort();
```
https://github.com/apache/maven-integration-testing/commit/482756ec1ff4b80c233bda949d187d7953144d47

**Comment by jira-importer** [comment]
Resolve #6722

**Pull request merged**